wgengine/ifstatus_windows.go

```diff
@@ -84,24 +85,31 @@ func waitIfaceUp(iface tun.Device, timeout time.Duration, logf logger.Logf) erro
 		return nil
 	}
 
-	iw.logf("waiting for interface to come up...")
+	expires := time.Now().UTC().Add(timeout)
+	tmr := time.NewTicker(10 * time.Second)
+	defer tmr.Stop()
 
-	tmr := time.NewTimer(timeout)
+	for {
+		iw.logf("waiting for interface to come up...")
+
+		select {
+		case <-iw.sig:
+			iw.logf("interface is up")
+			return nil
+		case <-tmr.C:
+			break;
+		}
 
-	select {
-	case <-iw.sig:
-		tmr.Stop()
-		iw.logf("interface is up")
-		return nil
-	case <-tmr.C:
-		iw.logf("timeout expired")
-		// Last chance - check one more time
 		if iw.isUp() {
-			// May happen only if NotifyIpInterfaceChange doesn't work (unlikely)
-			// or if the interface went up in the same moment the timeout has expired (also unlikely)
-			iw.logf("interface is up after timeout expired")
+			// Very unlikely to happen - either NotifyIpInterfaceChange doesn't work
+			// or it came up in the same moment as tick. Indicate this in the log message.
+			iw.logf("[tick] interface is up")
 			return nil
 		}
-		return fmt.Errorf("timeout expired")
+
+		if expires.Before(time.Now().UTC()) {
```

Comment:
no need for the .UTC() part. It does nothing here.

the timezone is only for display purposes, but there's no display here.

Reply:
I usually use UTC to avoid issues when local clocks jump +- hour due to daylight saving time changes. It happens rarely though (twice a year), so ok, will remove it.

Reply:
But to be clear: adding `.UTC()` here won't even affect daylight savings time. This code will still be correct during the twice a year transitions.

The internal representation of a Go `time.Time` is effectively a time-zone free unix timestamp and a timezone for formatting purposes. But `Before` and `After` comparison methods ignore the timezone entirely and operate entirely on the unix time part of the value.

Reply:
Oh, ok, nice to know! Thanks!